### ISSUE TITLE
html5 <video> tag support

### ISSUE THREAD

**Issue by chemel** [ezyang/htmlpurifier#106]
htmlpurifier remove the <video> tag, could you allow this tag?

Maybe you can draw inspiration from this piece of code: https://github.com/kennberg/php-htmlpurfier-html5/blob/master/htmlpurifier_html5.php#L59

Related issue: https://github.com/Exercise/HTMLPurifierBundle/issues/34

Thanks for this geat lib.

**Comment by ezyang** [comment]
Yes, in principle we should accept this tag, but we'll need to carefully audit it for XSS possibilities. People who want to live dangerously can perhaps just add video manually for now.

**Comment by Global90** [comment]
how to add video manually?? i use in Yii2

**Comment by jackewitz** [comment]
We just try to allow video-tags in limesurvey using htmlpurifier to prevent XSS-attacks. So the question above is highly responsible for us. Is there an answer to "how to add video manually"?

**Comment by ezyang** [comment]
For a hacky but probably will work solution, look at http://htmlpurifier.org/docs/enduser-youtube.html

If you want to support it as a proper tag, look at http://htmlpurifier.org/docs/enduser-customize.html

**Comment by jackewitz** [comment]
Thanks for your reply. Unfortunately: LimeSurvey use an array to configure htmlpurifier. Is it possible to add video tag via the configuration array (not via the configuration object)?

We tried it with

* HTML.Allowed
* HTML.AllowedElements and HTML.AllowedAttributes

but nothing works.

**Comment by lj3lj3** [comment]
I just follow the `enduser-customize.html ` and use bellow code, video tag works as expected! Maybe need to improve the some parameter, hope it helps.
```
$def = $htmlPurifierConfig->getHTMLDefinition(true);
$def->addElement(   // add video tag
    'video',   // name
    'Block',  // content set
    'Flow', // allowed children
    'Common', // attribute collection
    array( // attributes
        'src' => 'URI',
        'width' => 'Length',
        'height' => 'Length',
        'style' => 'CDATA'
    )
);

$purifier = new HTMLPurifier($htmlPurifierConfig);
```

**Comment by mattford** [comment]
https://github.com/xemlock/htmlpurifier-html5 adds HTML5 definitions to this lib

**Comment by mbnoimi** [comment]
Is there still any hope to implement this feature request soon? I haven't the proper experience to contribute the code but I found it a vital thing in 2022
I faced a problem with it [in this post](https://github.com/nextcloud/news/discussions/1856)

**Comment by bytestream** [comment]
@mbnoimi it's implemented in https://github.com/xemlock/htmlpurifier-html5

**Comment by mbnoimi** [comment]
> @mbnoimi it's implemented in https://github.com/xemlock/htmlpurifier-html5

Sorry for my mistake. I didn't notice that this repository doesn't belong to NextCloud (I came from [this discussion](https://github.com/nextcloud/news/discussions/1856))
